examples/esp32/xively_demo/main/main.c

```diff
@@ -303,3 +348,46 @@ void xt_recv_mqtt_msg_callback( const xi_sub_call_params_t* const params )
             io_led_on();
     }
 }
+
+/******************************************************************************
+ *        OTA Firmware Update notifications from the Xively Client's BSP
+ *
+ * Note:
+ * These functions are declared as extern in the ESP32's BSP:
+ *     xively-client-c/src/bsp/platform/esp32/xi_bsp_io_fs_esp32.c
+ * It's a custom solution to showcase how the application can be made aware of the
+ * download status. Useful to act on peripherals, display download progress, etc.
+ *
+ * If you decide to remove these functions for your app, you'll also have to
+ * remove their calls and declarations in xi_bsp_io_fs_esp32.c
+ ******************************************************************************/
+void esp32_xibsp_notify_update_started( const char* filename, size_t file_size )
+{
+    printf( "\nDownload of file [%s] started. Size: [%d]", filename, file_size );
+    ota_download_file_size = file_size;
+}
+
+void esp32_xibsp_notify_chunk_written( size_t chunk_size, size_t offset )
+{
+    if ( ota_download_file_size <= 0 )
+    {
+        return;
+    }
+    ota_download_progress = ( ( chunk_size + offset ) * 100 ) / ota_download_file_size;
+    printf( "\n[%d%%] Firmware chunk of size [%d] written at offset [%d]",
+            ota_download_progress, chunk_size, offset );
+
+    { /* FW update light show */
+        for ( int i = 4; i > 0; i-- )
+        {
+            io_led_set( i % 2 );
+            vTaskDelay( 50 / portTICK_PERIOD_MS );
+        }
+    }
+}
+
+void esp32_xibsp_notify_update_applied()
+{
+    printf( "\nFirmware package download complete - Proceeding with reboot" );
+    /* xi_bsp_fwu_on_package_download_finished called from the fwu BSP */
```

Comment:
do we need this? couldn't you put this printf into the xi_bsp_fwu_on_package_download_finished function directly?

Reply:
I try to use only `xi_debug_logger/format` in the BSPs, never printf()s. I added this message so the user knows why the device is rebooting, regardless of their BSP_DEBUG configuration.

If you think it's making things more confusing, I'll be happy to get rid of it.

Reply:
ok, we can keep this but make it used via function pointers as stated below in one of my suggestions.